src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -2507,48 +2515,41 @@ public void handleModifyEvent(CatalogModifyEvent event) {}
         public void handlePostModifyEvent(CatalogPostModifyEvent event) {
             CatalogInfo source = event.getSource();
             source.accept(this);
-
```

Comment:
These changes do not relate to making the module "usable from external code". Looks like this flush and the other one in ``handleRemoveEvent`` are related to JDBCConfig (guessing by the history of changes).

@NielsCharlier you added this bit of code as part of https://osgeo-org.atlassian.net/browse/GEOS-9169, could you please have a look?

Reply:
I did not notice... right now there are 278 modified files, hard to comment further. @groldan can you clean up the PR again?